packages/salesforcedx-vscode-core/src/index.ts

```diff
@@ -793,3 +797,35 @@ export const showWarningNotification = (type: string, args: any[]) => {
   const showMessage = nls.localize(type, ...args);
   void vscode.window.showWarningMessage(showMessage);
 };
+
+const handleTheUnhandled = (): void => {
+  process.on('unhandledRejection', (reason: Error, promise: Promise<any>) => {
+    const collectedData: { message?: string; stackTrace?: string | undefined } = {};
+    // Attach a catch handler to the promise to handle the rejection
+    promise.catch(error => {
+      // Collect relevant data
+      if (error instanceof Error) {
+        collectedData.message = error.message;
+        collectedData.stackTrace = error.stack ?? 'No stack trace available';
+      } else if (typeof error === 'string') {
+        collectedData.message = error;
+      }
+    });
+    // Capture stack trace if available
+    collectedData.stackTrace ??= reason
+      ? reason.stack
+      : 'No stack trace available';
+
+    // make an attempt to isolate the first reference to one of our extensions from the stack
+    const fromExtension = collectedData.stackTrace?.split(os.EOL)
+        .filter(l => l.includes('at '))
+        .flatMap(l => l.split(path.sep))
+        .find(w => w.startsWith('salesforcedx-vscode'));
+
+    // Send detailed telemetry data
+    telemetryService.sendException(
+      `${fromExtension ? fromExtension + '-' : ''}unhandledRejection`,
```

Comment:
I'd lean more towards a standard single exception name and just having the extension info be in the data (I think it will be be default...extname, version, etc)